src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -408,6 +414,27 @@ public void IncrementLogLinesCount(string logLevel)
             }
         }
 
+        public void RecordLogMessage(DateTime timestamp, string logLevel, string logMessage, string spanId, string traceId)
```

Comment:
As other log implementations are likely going to use this method as well, I think we should add more validation here. It looks like there are checks for logLevel, and logMessage not being null in log4net only code paths?

Reply:
Yeah, I can move it to the Agent class.

Reply:
updated